experiments/src/http_client.rs

```diff
@@ -7,16 +7,79 @@
 //! Working on the real Nimbus schema.
 
 use super::Experiment;
+use crate::Config;
 use anyhow::Result;
+use serde::de::SeqAccess;
+use serde_derive::Deserialize;
+use std::marker::PhantomData;
 use url::Url;
 use viaduct::{status_codes, Request, Response};
 
+const DEFAULT_BASE_URL: &str = "https://settings.stage.mozaws.net"; // TODO: Replace this with prod
+const DEFAULT_BUCKET_NAME: &str = "main";
+const DEFAULT_COLLECTION_NAME: &str = "messaging-experiments";
+
 // Making this a trait so that we can mock those later.
 pub(crate) trait SettingsClient {
     fn get_experiements_metadata(&self) -> Result<String>;
     fn get_experiments(&self) -> Result<Vec<Experiment>>;
 }
 
+#[derive(Deserialize)]
+struct ExperimentResponse {
+    #[serde(deserialize_with = "skip_invalid")]
+    data: Vec<Experiment>,
+}
+/// A custom deserializer that skips any invalid or null elements
+///
+/// This is needed to filter out experiments that do not abide by the
+/// schema. For production, we would probably need to remove this
+/// and ensure that the server contains *only* valid experiments
+fn skip_invalid<'de, D, T>(deserializer: D) -> Result<Vec<T>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+    T: serde::Deserialize<'de>,
+{
+    struct SkipInvalid<T>(PhantomData<T>);
+
+    impl<'de, T> serde::de::Visitor<'de> for SkipInvalid<T>
+    where
+        T: serde::Deserialize<'de>,
+    {
+        type Value = Vec<T>;
+
+        fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+            formatter.write_str("Array with invalid elements")
+        }
+
+        fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+        where
+            A: SeqAccess<'de>,
+        {
+            let mut vec = Vec::new();
+            // NOTE: In here we ignore any errors that appear in
+            // the deserialization process and instead
+            // skip any invalid elements
+            // In the future (when we have a source of *only* valid experiments)
```

Comment:
I think in the future there will probably be malformed experiments. should we tweak the comment?